tidy3d/components/simulation.py

```diff
@@ -1742,3 +1743,30 @@ def make_eps_data(coords: Coords):
         # combine all data into dictionary
         coords = sub_grid[coord_key]
         return make_eps_data(coords)
+
+    @classmethod
+    def from_file(cls, fname: str, **parse_kwargs):
+        """Loads a :class:`Tidy3dBaseModel` from .yaml or .json file.
+
+        Parameters
+        ----------
+        fname : str
+            Full path to the .yaml or .json file to load the :class:`Tidy3dBaseModel` from.
+        **parse_kwargs
+            Keyword arguments passed to either pydantic's ``parse_file`` or ``parse_raw`` methods
+            for ``.json`` and ``.yaml`` file formats, respectively.
+        Returns
+        -------
+        :class:`Tidy3dBaseModel`
+            An instance of the component class calling `load`.
+
+        Example
+        -------
+        >>> simulation = Simulation.from_file(fname='folder/sim.json')
+        """
+        try:
+            updater = Updater.from_file(fname)
+            sim_dict = updater.update_to_current()
+            return Simulation.parse_obj(sim_dict, **parse_kwargs)
```

Comment:
`log.warning("Could not update simulation to latest version! Trying to load as is.")` or something?